app/src/main/java/com/example/characters/adapter/UserAdapter.kt

```diff
@@ -9,17 +9,18 @@ import androidx.recyclerview.widget.RecyclerView
 import com.example.characters.databinding.FragmentUserBinding
 import com.example.characters.databinding.ItemLoadingBinding
 import com.example.characters.model.PageItem
+import com.example.characters.model.User
 
 class UserAdapter(
     context: Context,
-    private val click: (PageItem.User) -> Unit
-) : ListAdapter<PageItem, RecyclerView.ViewHolder>(DIFF_UTIL) {
+    private val click: (PageItem.Context<out User>) -> Unit
```

Comment:
здесь ты знаешь на какой элемент у тебя происходит клик и нет смысла обрабатывать его как sealed class
```suggestion
    private val onUserClicked: (User) -> Unit
```